src/selectors/disabled.js

```diff
@@ -34,18 +33,23 @@ export const getNodeDisabledPipeline = createSelector(
  * Calculate whether nodes should be disabled based on their tags
  */
 export const getNodeDisabledTag = createSelector(
-  [getNodeIDs, getTagEnabled, getTagCount, getNodeTags],
-  (nodeIDs, tagEnabled, tagCount, nodeTags) =>
-    arrayToObject(nodeIDs, nodeID => {
-      if (tagCount.enabled === 0) {
-        return false;
-      }
-      if (nodeTags[nodeID].length) {
-        // Hide task nodes that don't have at least one tag filter enabled
-        return !nodeTags[nodeID].some(tag => tagEnabled[tag]);
-      }
-      return true;
-    })
+  [getNodeIDs, getTagEnabled, getNodeTags],
+  (nodeIDs, tagEnabled, nodeTags) => {
+    const someEnabled = Object.values(tagEnabled).some(
+      enabled => enabled === true
+    );
+    return arrayToObject(
+      nodeIDs,
+      nodeID =>
+        someEnabled &&
+        (nodeTags[nodeID].length === 0 ||
+          nodeTags[nodeID].every(
+            tag =>
+              typeof tagEnabled[tag] === 'undefined' ||
+              tagEnabled[tag] === false
+          ))
```

Comment:
honestly, this is pretty difficult to read, so I'd prefer to break it up a bit, even if that means adding some if statements like it was before

Reply:
Yeah it's strange, prettier decided to make it into a staircase which doesn't help. Anyway, I've actually reverted this back to the original method now since it ended up coming full circle in the end.